app/CLI/Watch.hs

```diff
@@ -0,0 +1,50 @@
+module CLI.Watch (
+  watchAuction,
+) where
+
+-- Prelude
+import Hydra.Prelude (MonadDelay (threadDelay))
+import Prelude
+
+-- Haskell imports
+import Data.Time.Clock (getCurrentTime)
+import Data.Time.Format (defaultTimeLocale, formatTime)
+import System.Console.ANSI
+
+-- Hydra auction
+import CLI.Config (
+  AuctionName (..),
+  CliEnhancedAuctionTerms (..),
+  readCliEnhancedAuctionTerms,
+ )
+import HydraAuction.OnChain.Common (secondsLeftInInterval, stageToInterval)
+import HydraAuction.Tx.Common (currentAuctionStage, currentTimeMilliseconds)
+
+watchAuction :: AuctionName -> IO ()
+watchAuction auctionName = do
+  clearScreen
+  setCursorPosition 0 0
+
+  mEnhancedTerms <- readCliEnhancedAuctionTerms auctionName
+
+  case mEnhancedTerms of
+    Nothing ->
+      putStrLn $ "Auction " <> show auctionName <> " does not exist."
+    Just CliEnhancedAuctionTerms {sellerActor, terms} -> do
+      currentTime <- getCurrentTime
+      currentTimeMs <- currentTimeMilliseconds
+      currentStage <- currentAuctionStage terms
+      let showTime = formatTime defaultTimeLocale "%Y-%m-%d %H:%M:%S"
+          mSecsLeft = secondsLeftInInterval currentTimeMs (stageToInterval terms currentStage)
+      putStrLn $ showTime currentTime
+      putStrLn $ "Auction: " <> show auctionName
+      putStrLn $ "Seller: " <> show sellerActor
+      putStrLn $
+        "Current stage: "
+          <> show currentStage
+          <> case mSecsLeft of
+            Nothing -> mempty
```

Comment:
this can perhaps be an error message, telling that the `Interval` was not constructed properly by the `stageToInterval` function

Reply:
That is not the case though, the idea is that once you are in the `CleanupStage` there is no upperbound to the validity interval, and no amount of seconds left for the next stage.

Reply:
and that's the only codepath in all situations that can lead to `Nothing` there?